src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -219,8 +219,8 @@ public ResponseEntity<Map<String, FileType>> getFilesForPlagiarismView(@PathVari
      */
     @GetMapping(value = "repository-files-content/{commitId}", produces = MediaType.APPLICATION_JSON_VALUE)
     @EnforceAtLeastStudent
-    public ResponseEntity<Map<String, String>> getFilesAtCommit(@PathVariable String commitId, @RequestParam(required = false) Long participationId,
-            @RequestParam(required = false) RepositoryType repositoryType) {
+    public ResponseEntity<Map<String, String>> getFilesAtCommit(@PathVariable String commitId, @RequestParam(defaultValue = "0") long participationId,
```

Comment:
```suggestion
    public ResponseEntity<Map<String, String>> getFilesAtCommit(@PathVariable String commitId, @RequestParam() Optional<Long> participationId,
```
I think the id should not use 0 as default value

Reply:
After looking more into this function, it seems that this participationId is not actually optional as it is first passed into a method expecting a `long` rather than `Long` and then passed into `ArtemisJpaRepository::findByIdElseThrow`. For now I have made it explicitly throw the associated exception to make this clear, or how would you like this to be handled?